src/PrettyPrinting.jl

```diff
@@ -1591,23 +1591,89 @@ function _write_line(io::IOCustom, str::AbstractString)
     spaceleft = c - ind - io.printed
   end
   #@show spaceleft
-  firstlen = min(spaceleft, length(str))
-  firststr = str[1:firstlen]
-  if io.lowercasefirst
-    written += write(io.io, lowercasefirst(firststr))
-    io.lowercasefirst = false
+  _graphemes = Base.Unicode.graphemes(str)
+  firstlen = min(spaceleft, length(_graphemes))
+  # make an iterator over valid indices
+  firstiter = Base.Iterators.take(_graphemes, firstlen)
+  restiter = Base.Iterators.drop(_graphemes, firstlen)
+  firststr = join(firstiter)
+  if length(firstiter) == textwidth(firststr)
+    if io.lowercasefirst
+      written += write(io.io, lowercasefirst(firststr))
+      io.lowercasefirst = false
+    else
+      written += write(io.io, firststr)
+      io.lowercasefirst = false
+    end
+    io.printed += textwidth(firststr)
   else
-    written += write(io.io, firststr)
-    io.lowercasefirst = false
-  end
-  io.printed += textwidth(firststr)
-  reststr = str[firstlen + 1:end]
-  it = Iterators.partition(1:textwidth(reststr), c - ind > 0 ? c - ind : c)
-  for i in it
+    #firstline is wider than number of graphemes
+    partcollect = collect(firstiter)
+    printstr = ""
+    j = 1
+    while textwidth(printstr) < (c - ind > 0 ? c - ind : c)
+      printstr *= partcollect[j]
+      j += 1
+      if j > length(partcollect)
+        break
+      end
+    end
+    if io.lowercasefirst
+      written += write(io.io, lowercasefirst(printstr))
+    else
+      written += write(io.io, printstr)
+    end
+    io.printed += textwidth(printstr)
+
+    #the spillover string
     written += write(io.io, "\n")
     written += write_indent(io)
-    written += write(io.io, reststr[i])
-    io.printed = textwidth(reststr[i])
+    printstr = join(collect(firstiter)[j:end])
+    if io.lowercasefirst
+      written += write(io.io, lowercasefirst(printstr))
+      io.lowercasefirst = false
+    else
+      written += write(io.io, printstr)
+      io.lowercasefirst = false
+    end
+    io.printed += textwidth(printstr)
+  end
+  it = Iterators.partition(1:length(restiter), c - ind > 0 ? c - ind : c)
+  for i in it
+    # partitions of the spillover text
+    partcollect = collect(restiter)[i]
+    partstr = join(collect(restiter)[i])
```

Comment:
Doing `collect(restiter)` in each loop iteration (and twice at that!) will each time allocate a new `Vector`. Better to do it once before the loop
```suggestion
  rest = collect(restiter)
  for i in it
    # partitions of the spillover text
    partcollect = rest[i]
    partstr = join(rest[i])
```